app/main/posts/views/share/post-export.directive.js

```diff
@@ -28,54 +29,85 @@ function PostExportController(
     $q,
-    PostFilters
+    PostFilters,
+    _
 ) {
     $scope.loading = false;
-    $scope.exportPosts = exportPosts;
-
-    function exportPosts() {
+    this.getQuery = getQuery;
+    this.prepareExport = prepareExport;
+    this.showCSVResults = showCSVResults;
+    this.requestExport = requestExport;
+    $scope.exportPostsConfirmation = function () {
+        /**
+         * Trigger confirm notification for user.
+         * When the user accepts, get the CSV
+         */
         Notify.confirm('notify.post.export').then(function (message) {
+            prepareExport();
+        });
+    };
+    function prepareExport() {
+        loadingStatus(true);
+        var site = ConfigEndpoint.get({ id: 'site' }).$promise;
+        var query = getQuery();
+        var exportQuery = PostEndpoint.export(query);
+        requestExport(site, query, exportQuery);
+    }
+
+    function requestExport(site, query, exportQuery) {
+        $q.all([site, exportQuery]).then(function (response) {
+            showCSVResults(response, query.format);
+        }).finally(function (response) {
+            loadingStatus(false);
+        });
+    }
+    function loadingStatus(status) {
+        if (status === true) {
             $scope.loading = true;
             Notify.notifyProgress('<br><h3 translate="notify.export.in_progress">Your CSV export is in progress...</h3><br>');
-            if (!$scope.filters) {
-                $scope.filters = [];
-            }
-
-            var format = 'csv',  //@todo handle more formats
-                // Prepare filters for export
-                query = angular.extend({}, PostFilters.getQueryParams($scope.filters), {
-                    format: format
-                }),
-
-                site = ConfigEndpoint.get({ id: 'site' }).$promise,
-                postExport = PostEndpoint.export(query);
-
-            // Save export data to file
-            $q.all([site, postExport]).then(function (response) {
+        } else {
+            $scope.loading = false;
+            Notify.notify('<h3 translate="notify.export.complete">Your CSV export is complete.</h3><p translate="notify.export.complete_data_found_message">The data from your export can be found in your browser\'s downloads<p>');
+        }
+    }
+    function getQuery() {
+        /**
+         * If the filters are not available, apply the defaults
+         */
+        if (!$scope.filters || _.isEmpty($scope.filters)) {
+            $scope.filters = PostFilters.getDefaults();
+        }
+        var format = 'csv';  //@todo handle more formats
+        // Prepare filters for export
+        var query = angular.extend({}, PostFilters.getQueryParams($scope.filters), {
+            format: format
+        });
+        return query;
+    }
 
-                var filename = response[0].name + '-' + (new Date()).toISOString().substring(0, 10) + '.' + format,
-                    data = response[1].data;
+    function showCSVResults(response, format) {
```

Comment:
This is not a block to merging but it would be good to add a test covering this function.

Reply:
agree